### PR TITLE
fix: ignore the case when comparing azure tags in service annotation

### PR DESCRIPTION
**Pull request by nilo19**
#### What type of PR is this?

/kind bug

#### What this PR does / why we need it:

fix: ignore the case when comparing azure tags in service annotation

#### Which issue(s) this PR fixes:

Fixes #

#### Does this PR introduce a user-facing change?

```release-note
fix: ignore the case when comparing azure tags in service annotation
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs

```
/triage accepted
/sig cloud-provider
/area provider/azure
/priority critical-urgent

### CODE REVIEW

File: staging/src/k8s.io/legacy-cloud-providers/azure/azure_loadbalancer.go [hunk 1]
```diff
@@ -2137,9 +2137,8 @@ func (az *Cloud) ensurePIPTagged(service *v1.Service, pip *network.PublicIPAddre
 	if _, ok := service.Annotations[ServiceAnnotationAzurePIPTags]; ok {
 		annotationTags = parseTags(service.Annotations[ServiceAnnotationAzurePIPTags])
 	}
-	for k, v := range annotationTags {
-		configTags[k] = v
-	}
+	configTags, _ = reconcileTags(configTags, annotationTags)
```
Comment (feiskyer) [inline staging/src/k8s.io/legacy-cloud-providers/azure/azure_loadbalancer.go:2140]: Could you add some unit tests for it?

Comment (nilo19) [reply staging/src/k8s.io/legacy-cloud-providers/azure/azure_loadbalancer.go:2140]: already covered

**Review by feiskyer — APPROVED** [review]
/lgtm
/approve

**Comment by cheftako** [comment]
Approved #104734 before I ran across this. Please let me know if you feel this should be merged instead.

**Comment by nilo19** [comment]
@cheftako please merge this one, thanks!

**Pull request merged**